src/api/app/models/kiwi/image.rb

```diff
@@ -51,6 +53,29 @@ def self.build_from_xml(xml_string, md5)
     end
     new_image
   end
+
+  def to_xml
+    kiwi_file = package.kiwi_image_file
+    return nil unless kiwi_file
+    kiwi_body = package.source_file(kiwi_file)
+
+    xml_repos = repositories.map(&:to_xml).join("\n")
+
+    doc = Nokogiri::XML::DocumentFragment.parse(kiwi_body)
+    doc.xpath("image/repository").remove
+
+    image = doc.at_css('image')
+    image.first_element_child.after(xml_repos)
+
+    # Reparser for pretty printing
+    Nokogiri::XML(doc.to_xml, &:noblanks).to_xml
+  end
+
+  def write_to_backend
+    package.save_file({ filename: package.kiwi_image_file, file: to_xml })
+    self.md5_last_revision = package.kiwi_file_md5
```

Comment:
I didn't find any check for the kiwi file md5sum in ``` save_file! ```. This should be checked before doing any writes. And especially before updating the ``` self.md5_last_revision ```